application/forms/Authentication/LoginForm.php

```diff
@@ -95,9 +104,66 @@ public function onSuccess()
             $user->setDomain(Config::app()->get('authentication', 'default_domain'));
         }
         $password = $this->getElement('password')->getValue();
+        $rememberMeIsChecked = (/*isset($_POST['rememberme']) &&*/ $_POST['rememberme'] == '1') ? true : false;
+
         $authenticated = $authChain->authenticate($user, $password);
         if ($authenticated) {
             $auth->setAuthenticated($user);
+            if($rememberMeIsChecked) {
+                var_dump("set cookies for this user");
+                $config = [
+                    'digest_alg' => 'sha512',
+                    'private_key_bits' => 2048,
+                    'private_key_type' => OPENSSL_KEYTYPE_RSA,
+                ];
+
+                // Create the keypair
+                $res = openssl_pkey_new($config);
+
+                // Extract the private key from $res to $privKey
+                openssl_pkey_export($res, $privKey);
+
+                // Extract the public key from $res to $pubKey
+                $pubKey = openssl_pkey_get_details($res);
+                $pubKey = $pubKey["key"];
+
+                $name = $user->getUsername();
+                $pswd = $password;
+
+                // Encrypt the username to $name_encrypted using the public key
+                openssl_public_encrypt($name, $name_encrypted, $pubKey);
+
+                // Encrypt the username to $pswd_encrypted using the public key
+                openssl_public_encrypt($pswd, $pswd_encrypted, $pubKey);
+
+                // change crypt in binary code
+                $bname = base64_encode($name_encrypted);
+                $bpswd = base64_encode($pswd_encrypted);
+
+                // make an Array and serialize it to use it as string
+                $data = [$bname,$bpswd,$pubKey];
+                $data = serialize($data);
```

Comment:
We should not use `serialize()` here because that opens an attack vector if it is somehow possible to inject or manipulate cookie values. `serialize()` is not the problem but `unserialize()` is. Please find out why and create an alternative with `implode()` and `explode()`.